src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CoseKeyHelpers.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+
+namespace System.Formats.Cbor.Tests
+{
+    public static class CborCoseKeyHelpers
+    {
+        public static byte[] ExportCosePublicKey(ECDsa ecDsa, HashAlgorithmName hashAlgName)
+        {
+            ECParameters ecParams = ecDsa.ExportParameters(includePrivateParameters: false);
+            using var writer = new CborWriter(CborConformanceLevel.Ctap2Canonical);
+            WriteECParametersAsCosePublicKey(writer, ecParams, hashAlgName);
+            return writer.GetEncoding();
+        }
+
+        public static (ECDsa ecDsa, HashAlgorithmName hashAlgName) ParseCosePublicKey(byte[] coseKey)
```

Comment:
Should this be called just ReadPublicKey? Cose is already in the name of the class